build.gradle

```diff
@@ -162,6 +162,13 @@ task generateApiRequestFormat(type: cz.habarta.typescript.generator.gradle.Gener
     outputFile = "src/web/types/api-request.ts"
 }
 
+task generateApiEndPoints(type: cz.habarta.typescript.generator.gradle.GenerateTask) {
+    classPatterns = [
+            "teammates.ui.webapi.endpoints.**"
+    ]
+    outputFile = "src/web/types/api-endpoints.ts"
```

Comment:
Looks like you are not using `api-endpoints.ts` anywhere?

Reply:
Yup, I was just thinking about how to standardize the endpoints on the frontend and backend, but I forgot to remove it. I will remove it.

On a related note, it seems the current typescript generator can only convert Java classes to Typescript interfaces, so I can't use it to convert the existing endpoints in `Const.java` to a `string : string` mapping in the  `api-endpoints.ts` file. I've been searching for gradle plugins that can do that, but there doesn't seem to be any, so I am considering writing a gradle task to do that. Do you have any suggestions?

Reply:
Ok after some more searching I think I found [something that might work](https://github.com/vojtechhabarta/typescript-generator/issues/141). I'll give it a try first

Reply:
Yup seems to be working! I created a new gradle task called `generateApiEndpoints` that will convert the enum values in `src/main/java/teammates/ui/webapi/endpoints/ResourceEndpoints.java` to a typescript enum in `api-endpoints.ts` called `ResourceEndpoints`.

This is how the `ResourceEndpoints.java` file looks like:
```java
package teammates.ui.webapi.endpoints;

import com.fasterxml.jackson.annotation.JsonValue;
import teammates.common.util.Const.ResourceURIs;

public enum ResourceEndpoints {
    URI_PREFIX(ResourceURIs.URI_PREFIX),
    LOGOUT(ResourceURIs.LOGOUT),
    DATABUNDLE(ResourceURIs.DATABUNDLE),
    EXCEPTION(ResourceURIs.EXCEPTION),
    ERROR_REPORT(ResourceURIs.ERROR_REPORT),
    AUTH(ResourceURIs.AUTH),
    ACCOUNT(ResourceURIs.ACCOUNT),
    ACCOUNTS_SEARCH(ResourceURIs.ACCOUNTS_SEARCH),
    ACCOUNT_RESET(ResourceURIs.ACCOUNT_RESET),
    ACCOUNT_DOWNGRADE(ResourceURIs.ACCOUNT_DOWNGRADE),
    RESPONSE_COMMENT(ResourceURIs.RESPONSE_COMMENT),
    COURSE(ResourceURIs.COURSE),
    COURSE_ARCHIVE(ResourceURIs.COURSE_ARCHIVE),
    BIN_COURSE(ResourceURIs.BIN_COURSE),
    COURSE_SECTIONS(ResourceURIs.COURSE_SECTIONS),
    COURSES(ResourceURIs.COURSES),
    INSTRUCTORS(ResourceURIs.INSTRUCTORS),
    INSTRUCTOR(ResourceURIs.INSTRUCTOR),
    INSTRUCTOR_PRIVILEGE(ResourceURIs.INSTRUCTOR_PRIVILEGE),
    RESULT(ResourceURIs.RESULT),
    STUDENTS(ResourceURIs.STUDENTS),
    STUDENT(ResourceURIs.STUDENT),
    SESSIONS_ONGOING(ResourceURIs.SESSIONS_ONGOING),
    SESSION(ResourceURIs.SESSION),
    SESSION_PUBLISH(ResourceURIs.SESSION_PUBLISH),
    SESSION_REMIND_SUBMISSION(ResourceURIs.SESSION_REMIND_SUBMISSION),
    SESSION_REMIND_RESULT(ResourceURIs.SESSION_REMIND_RESULT),
    SESSION_STATS(ResourceURIs.SESSION_STATS),
    SESSION_SUBMITTED_GIVER_SET(ResourceURIs.SESSION_SUBMITTED_GIVER_SET),
    SESSIONS(ResourceURIs.SESSIONS),
    BIN_SESSION(ResourceURIs.BIN_SESSION),
    QUESTIONS(ResourceURIs.QUESTIONS),
    QUESTION_RECIPIENTS(ResourceURIs.QUESTION_RECIPIENTS),
    RESPONSE(ResourceURIs.RESPONSE),
    RESPONSES(ResourceURIs.RESPONSES),
    HAS_RESPONSES(ResourceURIs.HAS_RESPONSES),
    SUBMISSION_CONFIRMATION(ResourceURIs.SUBMISSION_CONFIRMATION),
    JOIN(ResourceURIs.JOIN),
    JOIN_REMIND(ResourceURIs.JOIN_REMIND),
    TIMEZONE(ResourceURIs.TIMEZONE),
    LOCAL_DATE_TIME(ResourceURIs.LOCAL_DATE_TIME),
    SESSION_LINKS_RECOVERY(ResourceURIs.SESSION_LINKS_RECOVERY),
    NATIONALITIES(ResourceURIs.NATIONALITIES),
    STUDENT_PROFILE_PICTURE(ResourceURIs.STUDENT_PROFILE_PICTURE),
    STUDENT_PROFILE(ResourceURIs.STUDENT_PROFILE),
    STUDENTS_CSV(ResourceURIs.STUDENTS_CSV),
    STUDENTS_AND_FEEDBACK_SESSION_DATA_SEARCH(ResourceURIs.STUDENTS_AND_FEEDBACK_SESSION_DATA_SEARCH),
    COURSE_ENROLL_STUDENTS(ResourceURIs.COURSE_ENROLL_STUDENTS);

    ResourceEndpoints(String s) {
        this.url = s;
    }
    private final String url;

    @JsonValue
    public String getUrl() {
        return url;
    }
}

```

And this is how the `api-endpoints.ts` file looks like:

```typescript
export enum ResourceEndpoints {
  URI_PREFIX = "/webapi",
  LOGOUT = "/logout",
  DATABUNDLE = "/databundle",
  EXCEPTION = "/exception",
  ERROR_REPORT = "/errorreport",
  AUTH = "/auth",
  ACCOUNT = "/account",
  ACCOUNTS_SEARCH = "/accounts/search",
  ACCOUNT_RESET = "/account/reset",
  ACCOUNT_DOWNGRADE = "/account/downgrade",
  RESPONSE_COMMENT = "/responsecomment",
  COURSE = "/course",
  COURSE_ARCHIVE = "/course/archive",
  BIN_COURSE = "/bin/course",
  COURSE_SECTIONS = "/course/sections",
  COURSES = "/courses",
  INSTRUCTORS = "/instructors",
  INSTRUCTOR = "/instructor",
  INSTRUCTOR_PRIVILEGE = "/instructor/privilege",
  RESULT = "/result",
  STUDENTS = "/students",
  STUDENT = "/student",
  SESSIONS_ONGOING = "/sessions/ongoing",
  SESSION = "/session",
  SESSION_PUBLISH = "/session/publish",
  SESSION_REMIND_SUBMISSION = "/session/remind/submission",
  SESSION_REMIND_RESULT = "/session/remind/result",
  SESSION_STATS = "/session/stats",
  SESSION_SUBMITTED_GIVER_SET = "/session/submitted/giverset",
  SESSIONS = "/sessions",
  BIN_SESSION = "/bin/session",
  QUESTIONS = "/questions",
  QUESTION_RECIPIENTS = "/question/recipients",
  RESPONSE = "/response",
  RESPONSES = "/responses",
  HAS_RESPONSES = "/hasResponses",
  SUBMISSION_CONFIRMATION = "/submission/confirmation",
  JOIN = "/join",
  JOIN_REMIND = "/join/remind",
  TIMEZONE = "/timezone",
  LOCAL_DATE_TIME = "/localdatetime",
  SESSION_LINKS_RECOVERY = "/sessionlinksrecovery",
  NATIONALITIES = "/nationalities",
  STUDENT_PROFILE_PICTURE = "/student/profilePic",
  STUDENT_PROFILE = "/student/profile",
  STUDENTS_CSV = "/students/csv",
  STUDENTS_AND_FEEDBACK_SESSION_DATA_SEARCH = "/studentsAndSessionData/search",
  COURSE_ENROLL_STUDENTS = "/course/enroll/students",
}
```

Reply:
Nice work!